graph-builder/src/graph.rs

```diff
@@ -142,5 +150,101 @@ fn create_graph(
             })
         })?;
 
+    graph.prune_abstract();
+
     Ok(graph)
 }
+
+/// Removes and retrieves the metadata value of the given `key` which is prefixed
+/// by `filter`
+fn remove_prefixed_label(
+    filter: &str,
+    release: &mut registry::Release,
+    key: &str,
+) -> Option<String> {
+    release
+        .metadata
+        .metadata
+        .remove(&format!("{}.{}", filter, key))
+}
+
+/// Remove releases which are labeled for removal.
+///
+/// The labels are assumed to have the syntax `{filter}.release-remove=<bool>`
+fn process_release_remove_label(
+    filter: &str,
+    release: registry::Release,
+) -> Option<registry::Release> {
+    let mut release = release;
+
+    if let Some(s) = remove_prefixed_label(filter, &mut release, "release.remove") {
+        // Filter releases which have "{prefix}.release-remove=true"
+        let remove = std::str::FromStr::from_str(&s).unwrap_or_else(|e| {
+            debug!("could not parse '{}': {}", s, e);
```

Comment:
I'd go for `error` here, in the hope that a human sees and fixes this.
And also make it more explicit like `could not parse {} as a boolean (assuming true): {}`

I also wonder if we should be more conservative and default to `false` instead on errors.